src/Processor/CancelOrRefundProcessor.php

```diff
@@ -0,0 +1,57 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Webhook Module for PHP
+ *
+ * Copyright (c) 2021 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ */
+
+namespace Adyen\Webhook\Processor;
+
+use Adyen\Webhook\EventCodes;
+use Adyen\Webhook\PaymentStates;
+
+class CancelOrRefundProcessor extends Processor implements ProcessorInterface
+{
+    const MODIFICATION_ACTION = 'modification.action';
+    const CANCEL = 'cancel';
+    const REFUND = 'refund';
+
+    public function process(): ?string
+    {
+        $state = $this->initialState;
+        $logContext = [
+            'eventCode' => EventCodes::CANCEL_OR_REFUND,
+            'originalState' => $state
+        ];
+
+        if ($this->notification->isSuccess() && isset($this->notification->additionalData[self::MODIFICATION_ACTION])) {
+            if ($this->notification->additionalData[self::MODIFICATION_ACTION] === self::CANCEL) {
+                $state = PaymentStates::STATE_CANCELLED;
```

Comment:
I think we're missing the initial state check here. For example, if the initial state is PENDING and modification.action is REFUND we would return REFUNDED, but should be CANCELLED instead. Or am I missing something?

Reply:
I was thinking that receiving the cancel or refund the initial state it is not important as in any case the MODIFICATION_ACTION is the one to move the state

Reply:
I think we need to validate this assumption just to be sure